app/services/user_access_key.rb

```diff
@@ -12,11 +11,21 @@
 class UserAccessKey
   attr_accessor :encrypted_d, :salt, :z1, :z2, :random_r
```

Comment:
I don't see `encrypted_d` being set anywhere but I see that a test expects it to equal `subject.xor(ciphertext)`. Where does that happen? Does this come from some gem? Perhaps it might be worth adding a comment?